lib/livebook/intellisense.ex

```diff
@@ -519,14 +522,33 @@ defmodule Livebook.Intellisense do
     """
   end
 
+  defp format_definition_link(module, function_or_type \\ nil) do
+    path =
+      :code.get_path()
+      |> Enum.find(&(to_string(&1) =~ "livebook_runtime"))
+      |> to_string()
```

Comment:
It would be nice if we could already pass the directory as part of the context here: https://github.com/livebook-dev/livebook/blob/953f8cdddb200f96b7560d8308081587628af4d5/lib/livebook/intellisense.ex#L21 then we don't need to look at :code.get_path :)

Reply:
Another idea is to already generate the link to the exact cell and line here, something like this:

```
goto:CELLNAME?line=3
```

This means that, once the link is clicked, you don't have to go the server, you know which cell and line to go to. I think this can be done in two ways:

1. Store all identifier metadata in a ETS table and we have one identifier metadata table per session. Then this code can lookup the ETS table.

2. Extract the identifier building logic to a shared module you can use here. Then you can lookup the .beam file directly from here and extract the correct cell and line.